src/providers/local/staking.provider.ts

```diff
@@ -0,0 +1,395 @@
+          shares: getBptBalanceFiatValue(pool, unstakedBpt),
+          bpt: unstakedBpt
+        };
+      });
+      return decoratedPools;
+    });
+
+    /**
+     * METHODS
+     */
+    async function stakeBPT() {
+      if (!poolAddress.value) {
+        throw new Error(
+          `Attempted to call stake, however useStaking was initialised without a pool address.`
+        );
+      }
+      const gaugeAddress = await getGaugeAddress(poolAddress.value);
+      const gauge = new LiquidityGauge(gaugeAddress, getProvider());
+      const tx = await gauge.stake(
+        parseUnits(balanceFor(getAddress(poolAddress.value)), 18)
+      );
+      return tx;
+    }
+
+    async function unstakeBPT() {
+      if (!poolAddress.value) {
+        throw new Error(
+          `Attempted to call unstake, however useStaking was initialised without a pool address.`
+        );
+      }
+      const gaugeAddress = await getGaugeAddress(getAddress(poolAddress.value));
+      const gauge = new LiquidityGauge(gaugeAddress, getProvider());
+      const tx = await gauge.unstake(
+        parseUnits(stakedSharesForProvidedPool.value || '0', 18)
+      );
+      return tx;
+    }
+
+    async function getStakedShares() {
+      if (!poolAddress.value) {
+        throw new Error(
+          `Attempted to get staked shares, however useStaking was initialised without a pool address.`
+        );
+      }
+      const gaugeAddress = await getGaugeAddress(getAddress(poolAddress.value));
+      const gauge = new LiquidityGauge(gaugeAddress, getProvider());
+      const balance = await gauge.balance(account.value);
+      return formatUnits(balance.toString(), 18);
+    }
+
+    async function getGaugeAddress(poolAddress: string): Promise<string> {
+      const gaugeInterface = new Interface(GaugeFactoryABI);
+      const contract = new Contract(
+        configService.network.addresses.gaugeFactory,
+        gaugeInterface,
+        getProvider()
+      );
+      const gaugeAddress = await contract.getPoolGauge(getAddress(poolAddress));
+      return gaugeAddress;
+    }
+
+    function setPoolAddress(address: string) {
+      _poolAddress.value = address;
+    }
+
+    provide(StakingProviderSymbol, {
+      userGaugeShares,
+      userLiquidityGauges,
+      stakedSharesForProvidedPool,
+      stakedPools,
+      isLoadingStakingData,
+      isLoadingStakedPools,
+      isLoading,
+      isLoadingStakedShares,
+      isLoadingPoolEligibility,
+      isStakeDataIdle,
+      isStakedSharesIdle,
+      isRefetchingStakedShares,
+      isPoolEligibleForStaking,
+      refetchStakedShares,
+      isStakedPoolsQueryEnabled,
+      hideAprInfo: true,
+      isLoadingUserPools,
+      isUserPoolsIdle,
+      getGaugeAddress,
+      stakeBPT,
+      unstakeBPT,
+      getStakedShares,
+      setPoolAddress,
+      refetchStakingData
+    });
+  },
+
+  render() {
+    return h('div', this.$slots?.default ? this.$slots.default() : []);
```

Comment:
what is this doing?

Reply:
this is needed to render children for the provider